components/datasciencepipelines/datasciencepipelines.go

```diff
@@ -141,3 +149,20 @@ func (d *DataSciencePipelines) ReconcileComponent(ctx context.Context,
 
 	return nil
 }
+
+func argoWorkflowDeploymentAllowed(ctx context.Context,
+	cli client.Client) error {
+	workflowCRD := &apiextensionsv1.CustomResourceDefinition{}
+	if err := cli.Get(ctx, client.ObjectKey{Name: "workflows.argoproj.io"}, workflowCRD); err != nil {
+		if apierrs.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get Workflow CRD : %w", err)
+	}
+	// Verify if existing workflow is deployed by ODH
+	_, odhLabelExists := workflowCRD.Labels[labels.ODH.Component(ComponentName)]
+	if odhLabelExists {
+		return nil
+	}
+	return fmt.Errorf("failed to deploy DSP. Workflow CRD already exists. Remove CRD to upgrade")
```

Comment:
```suggestion
	return fmt.Errorf("failed to deploy DSP. Argo Workflow CRD already exists but not deployed by our operator . Remove CRD to upgrade")
```

Reply:
nitpick: replace "our" by "this" or "ODH".